static/app/components/dashboards/widgetQueriesForm.tsx

```diff
@@ -189,9 +189,29 @@ class WidgetQueriesForm extends React.Component<Props> {
           organization={organization}
           onChange={fields => {
             const fieldStrings = fields.map(field => generateFieldAsString(field));
+            const aggregateAliasFieldStrings = fieldStrings.map(field =>
```

Comment:
Hmm do you think it might be easier to handle this in widgetQueryFields - when a field is removed or changed, we check if it's being used as a sort and if it is, we can update the sort? So we don't have to end up looping through and constructing field aliases for each all columns each time something is changed?

Reply:
Hey Shruthi, took a look at this and `widgetQueryFields` isn't exposed to the widget query and the order by ie it only takes the the fields of the query 😞 . We could refactor to expose orderby in `widgetQueryFields` but I think it's a bit nicer to keep `widgetQueryFields` isolated. Do you think this is a blocker for this PR? If so I can dig a bit to see if I can cleanup up this management in some way.

Reply:
Ah, I see! Not a blocker, but I think we could consider passing a callback like 'updateOrderBy' or something? If you think it's something we should just do as a follow up, I'm okay with it. I don't want to hold up this bug fix!

Reply:
Thanks Shruthi! Will investigate reworking some of this in another pr so we can get this fix in!